yt_dlp/extractor/zenporn.py

```diff
@@ -0,0 +1,128 @@
+        'url': 'https://zenporn.com/video/8531117/amateur-students-having-a-fuck-fest-at-club/',
+        'md5': '67411256aa9451449e4d29f3be525541',
+        'info_dict': {
+            'id': '8531117',
+            'extr_id': '12791908',
+            'ext': 'mp4',
+            'title': 'Amateur students having a fuck fest at club',
+            'description': '',
+            'thumbnail': 'https://tn.txxx.tube/contents/videos_screenshots/12791000/12791908/288x162/1.jpg',
+            'post_date': '20191005',
+            'uploader': 'Jackopenass',
+            'age_limit': 18
+        }
+    }, {
+        'url': 'https://zenporn.com/video/15872038/glad-you-came/',
+        'md5': '296ccab437f5bac6099433768449d8e1',
+        'info_dict': {
+            'id': '15872038',
+            'extr_id': '111585',
+            'ext': 'mp4',
+            'title': 'Glad You Came',
+            'description': '',
+            'thumbnail': 'https://vpim.m3pd.com/contents/videos_screenshots/111000/111585/480x270/1.jpg',
+            'post_date': '20231024',
+            'uploader': 'Martin Rudenko',
+            'age_limit': 18
+        }
+    }]
+
+    def _gen_info_url(self, ext_domain, extr_id, lifetime=86400):
+        """ This function is a reverse engineering from the website javascript """
+        result = '/'.join(str(int_or_none(extr_id) // i * i) for i in (1_000_000, 1_000, 1))
+        return f'https://{ext_domain}/api/json/video/{lifetime}/{result}.json'
+
+    @staticmethod
+    def _decode_video_url(encoded_url):
+        """ This function is a reverse engineering from the website javascript """
+        # Replace lookalike characters and standardize map
+        translation = str.maketrans('АВСЕМ.,~', 'ABCEM+/=')
+        try:
+            return base64.b64decode(encoded_url.translate(translation), validate=True).decode()
+        except (binascii.Error, ValueError):
+            return None
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        ext_domain, extr_id = self._search_regex(
+            r'https://(?P<ext_domain>[\w.-]+\.\w{3})/embed/(?P<extr_id>\d+)/',
+            webpage, 'embed info', group=('ext_domain', 'extr_id'), fatal=True)
+
+        info_json = self._download_json(
+            self._gen_info_url(ext_domain, extr_id), video_id,
+            note='Downloading video info JSON', fatal=True)
+
+        video_json = self._download_json(
+            f'https://{ext_domain}/api/videofile.php', video_id, query={
+                'video_id': extr_id,
+                'lifetime': 8640000,
+            }, note='Downloading video metadata JSON', fatal=True)
+
+        encoded_url = video_json[0]['video_url']
+
+        decoded_url = self._decode_video_url(encoded_url)
+        if not decoded_url:
+            raise ExtractorError('Unable to decode the video url')
+
+        return {
+            'id': video_id,
+            'extr_id': extr_id,
```

Comment:
if we want to extract this value, we need to use a common info_dict field like `display_id`:
```suggestion
            'display_id': extr_id,
```
although arguably this should be the `id` and `video_id` should actually be `display_id`.

whichever you decide, tests will need to be updated

Reply:
Thanks for the input. I'll follow your suggestion.